packages/atomic/src/components/insight/user-actions/atomic-insight-user-action/atomic-insight-user-action.tsx

```diff
@@ -0,0 +1,97 @@
+import {FunctionalComponent, h} from '@stencil/core';
+import {UserAction} from '../..';
+import BookmarkIcon from '../../../../images/bookmark.svg';
+import DocumentIcon from '../../../../images/document.svg';
+import PointIcon from '../../../../images/point.svg';
+import QuickviewIcon from '../../../../images/quickview.svg';
+import SearchIcon from '../../../../images/search.svg';
+import {parseTimestampToDateDetails} from '../../../../utils/date-utils';
+import {InsightBindings} from '../../atomic-insight-interface/atomic-insight-interface';
+
+const icons = {
+  SEARCH: SearchIcon,
+  CLICK: DocumentIcon,
+  VIEW: QuickviewIcon,
+  CUSTOM: PointIcon,
+  TICKET_CREATION: BookmarkIcon,
+};
+
+interface AtomicInsightUserAction {
+  action: UserAction;
+  bindings: InsightBindings;
+}
+
+export const AtomicInsightUserAction: FunctionalComponent<
+  AtomicInsightUserAction
+> = ({bindings, action}) => {
+  const renderActionTimestamp = () => {
+    const {hours, minutes} = parseTimestampToDateDetails(action.timestamp);
+
+    const formattedHours = String(hours).padStart(2, '0');
+    const formattedMinutes = String(minutes).padStart(2, '0');
+
+    return `${formattedHours}:${formattedMinutes}`;
+  };
+
+  const renderIcon = () => {
+    const iconClasses = ['w-3', 'h-3'];
+    if (['CLICK', 'VIEW'].includes(action.actionType)) {
+      iconClasses.push('text-primary');
+    }
```

Comment:
I prefer this way? What do you think? 100% up to you, and we should do it the same way other Atomic code is written if you find other examples.

```suggestion
    const iconClasses = `w-3 h-3 ${['CLICK', 'VIEW'].includes(action.actionType) ? 'text-primary' : ''}`;
```